xmtp_api/src/mls.rs

```diff
@@ -307,6 +310,34 @@ where
 
         Ok(all_responses)
     }
+
+    pub async fn get_newest_message_metadata(
+        &self,
+        group_ids: Vec<&[u8]>,
+    ) -> Result<MessageMetadataMap> {
+        const BATCH_SIZE: usize = 1000;
+
+        let res =
+            futures::future::try_join_all(group_ids.chunks(BATCH_SIZE).map(|chunk| async move {
+                self.api_client
+                    .get_newest_group_message(GetNewestGroupMessageRequest {
+                        group_ids: chunk.to_vec().iter().map(|id| id.to_vec()).collect(),
+                        include_content: false,
+                    })
+                    .await
+                    .map_err(crate::dyn_err)
+            }))
+            .await?;
+
+        // Functionally process responses into metadata map
+        let metadata_map = res
+            .into_iter()
+            .flatten()
+            .filter_map(|response| response.map(|msg| (msg.group_id.clone(), msg)))
```

Comment:
nit: I don't think the `.clone()` is necessary here.

Reply:
No such luck, since the group_id comes from the msg.